backend/src/crawler/udn_crawler.py

```diff
@@ -0,0 +1,197 @@
+            "id": f"search:{quote(search_term)}",
+            "channelId": self.CHANNEL_ID,
+            "type": "searchword",
+        }
+
+    def _perform_request(self, url: str | None = None, params: dict | None = None) -> Response:
+        try:
+            response = requests.get(url, params=params)
+            response.raise_for_status()
+            return response
+        except RequestException as e:
+            raise RuntimeError(f"Failed to perform request to {url}: {e}")
+        
+    @staticmethod
+    def _parse_headlines(response: Response) -> list[Headline]:
+        try:
+            data = response.json()
+            if "lists" not in data:
+                raise ValueError("The response does not contain 'lists'")
+            processed_items = [
+                {"title": item["title"], "url": item["titleLink"]}
+                for item in data["lists"]
+            ]
+        
+            return [Headline(**item) for item in processed_items]
+        except (KeyError, ValueError, TypeError) as e:
+            raise RuntimeError(f"Failed to parse headlines: {e}")
+
+    def parse(self, url: str) -> News:
+        response = self._perform_request(url=url)
+        soup = BeautifulSoup(response.text, "html.parser")
+
+        return self._extract_news(soup, url)
+
+    @staticmethod
+    def _extract_news(soup: BeautifulSoup, url: str) -> News:
+        title = soup.find("h1", class_="article-content__title").text
+        time = soup.find("time", class_="article-content__time").text
+        content_section = soup.find("section", class_="article-content__editor")
+
+        paragraphs = [
+            p.text
+            for p in content_section.find_all("p")
+            if p.text.strip() != "" and "▪" not in p.text
+        ]
+        content = " ".join(paragraphs)
+
+        return News(
+            title=title,
+            url=url,
+            time=time,
+            content=content
+        )
+    
+    def add_news_summary(self, news: News, summary_result: str) -> NewsWithSummary:
+        """
+        Adds a summary and reason to a News object, returning a NewsWithSummary object.
+
+        :param news: News object containing basic news details.
+        :param summary_result: JSON string containing the summary and reason.
+        :return: NewsWithSummary object with added summary and reason.
+        """
+        try:
+            summary_data = json.loads(summary_result)
+            summary =  summary_data["影響"]
+            reason = summary_data["原因"]
+        except json.JSONDecodeError:
+            print(f"Failed to decode summary_result: {summary_result}")
+            summary = ""
+            reason = ""
+
+        news_with_summary = NewsWithSummary(
+            title=news.title,
+            url=news.url,
+            time=news.time,
+            content=news.content,
+            summary=summary,
+            reason=reason,
+        )
+        
+        return news_with_summary
```

Comment:
I noticed that there's a function named `add_news_summary(news: News, summary_result: str)` which calls this method in `backedn/src/news/utils.py`.

It seems to me that we can just move this method to `utils.py` since the crawler is mainly responsible for fetching, parsing, and saving news.